cli/internal/helm/client.go

```diff
@@ -109,9 +116,11 @@ func (c *Client) Upgrade(ctx context.Context, config *config.Config, timeout tim
 		if info == constellationOperatorsInfo || info == constellationServicesInfo {
```

Comment:
I don't think it makes sense to iterate over 4 values but have explicit checks for half of them inside the loop. I think we could refactor this for loop to its own function and call the function in 2 different for loops one passing the `config.MicroserviceVersion` and one `chart.Metadata.Version` to the function.

Reply:
I agree that this isn't the prettiest solution. But I don't yet see the improvement from your suggestion.

I assume you want to put the content of the for loop into a new function `new` and then have two for-loops, one with `ciliumInfo, certManagerInfo` and one with `constellationOperatorsInfo, constellationServicesInfo`. Both for-loops call `new`.

Two problems I see with that:
- The error handling still needs to happen in the caller. Therefore the [switch statement](https://github.com/edgelesssys/constellation/pull/1649/files/aec6a499733dec4266f7a3e3c0cc505d59987925#diff-9a96aebbcd10227f19a06b17202f088565d3cbd804f3b1c9ac4afd3b3ce08ab9R124-R131) needs to be duplicated, once for each loop
- Calling [updateVersions](https://github.com/edgelesssys/constellation/pull/1649/files/aec6a499733dec4266f7a3e3c0cc505d59987925#diff-9a96aebbcd10227f19a06b17202f088565d3cbd804f3b1c9ac4afd3b3ce08ab9R118) on all charts _should_ work. But I feel saver calling it just on our charts. Therefore there would be the same check inside the new function.

Reply:
We discussed the problem in a call and came up with this: https://github.com/edgelesssys/constellation/pull/1649/commits/0948dc32ab66798ca1d259ab128c74c1eb6f678a